Enterprise-CMCS/macpro-onemac
# OY2 23115: Dashboard Updates: Default Settings (hide columns based on role)

Story: https://qmacbis.atlassian.net/browse/OY2-23115
Endpoint: See github-actions bot comment

### Details
In order to keep the dashboard table from being too wide, certain columns are going to be hidden as a default.  This is based on role (currently State v CMS).

### Changes
- created a default hidden array for state and cms
- mapped the default hidden arrays to the roles
- put the default as the "default" if there is nothing in the session/local variables.

### Test Plan
1. Login as State user
2. Verify that "State" column defaults to hidden
3. Verify that "State" column can be added/removed as expected
4. Verify that no other functionality has changed with saving/showing the columns
5. Login as CMS user
6. Verify that "Submitted by" column defaults to hidden
7. Verify that "Submitted by" column can be added/removed as expected
8. Verify that no other functionality has changed with saving/showing the columns
